builtin/providers/google/resource_monitoring_group.go

```diff
@@ -0,0 +1,248 @@
+				Computed: true,
+			},
+
+			"displayName": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"parentName": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+				ForceNew: true,
+			},
+
+			"filter": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"isCluster": &schema.Schema{
+				Type:     schema.TypeBool,
+				Default:  false,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func getGroup(d *schema.ResourceData, config *Config) (*monitoring.Group, error) {
+	call := config.clientMonitoring.Projects.Groups.Get(d.Id())
+	group, err := call.Do()
+	if err != nil {
+		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
+			log.Printf("[WARN] Removing Group %q because it's gone", d.Get("displayName").(string))
+			// The resource doesn't exist anymore
+			id := d.Id()
+			d.SetId("")
+
+			return nil, fmt.Errorf("Resource %s no longer exists", id)
+		}
+
+		return nil, fmt.Errorf("Error reading group: %s", err)
+	}
+
+	return group, nil
+}
+
+func findGroup(d *schema.ResourceData, config *Config) (*monitoring.Group, error) {
+	var res *monitoring.ListGroupsResponse = nil
+	var listParam string = ""
+
+	// Determine if we're in a pagination recursion
+	nextPageToken, ok := d.GetOk("nextPageToken")
+	if !ok {
+		project, err := getProject(d, config)
+		if err != nil {
+			return nil, err
+		}
+		listParam = fmt.Sprintf("projects/%s", project)
+	} else {
+		listParam = nextPageToken.(string)
+	}
+	query := config.clientMonitoring.Projects.Groups.List(listParam)
+	// If a parent is configured set that on the query
+	v, ok := d.GetOk("parentName")
+	if ok {
+		query.ChildrenOfGroup(v.(string))
+	}
+	res, err := query.Do()
+	if err != nil {
+		return nil, err
+	}
+
+	// Check every group on this page for a matching display name
+	for _, element := range res.Group {
+		if element.DisplayName == d.Get("displayName") {
+			return element, nil
+		}
+	}
+	// If there's a next page token we'll need to recurse
+	if res.NextPageToken != "" {
+		d.Set("nextPageToken", res.NextPageToken)
+		return findGroup(d, config)
+	} // No next page, done checking and not found.
+	return nil, nil
+}
+
+func resourceMonitoringGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	// Build the group
+	group := &monitoring.Group{}
+
+	if v, ok := d.GetOk("displayName"); ok {
+		group.DisplayName = v.(string)
+	}
+
+	if v, ok := d.GetOk("parentName"); ok {
+		group.ParentName = v.(string)
+	}
+
+	if v, ok := d.GetOk("filter"); ok {
+		group.Filter = v.(string)
+	}
+
+	if v, ok := d.GetOk("isCluster"); ok {
+		group.IsCluster = v.(bool)
+	}
+
+	// Add the group
+	name := fmt.Sprintf("projects/%s", project)
+	call := config.clientMonitoring.Projects.Groups.Create(name, group)
+	res, err := call.Do()
+	if err != nil {
+		return err
+	}
+
```

Comment:
If you take this opportunity to call `d.SetId(res.Name)`, we can avoid listing all the groups later and matching on a non-unique index (the display name), which I get the feeling could come back to bite us.